lib/cext/preprocess.rb

```diff
@@ -145,6 +165,213 @@
+        replacement: 'rb_tr_managed_from_handle(wrapper->fields) \1'
+      },
+      {
+        match: /\bwrapper->fields\s*=\s*(\w.+);\s*$/ ,
+        replacement: 'wrapper->fields = rb_tr_handle_for_managed(\1);'
+      },
+      {
+        match: /xfree\(wrapper\);/,
+        replacement: MYSQL2_FREE_RESULT_WRAPPER
+      },
+      {
+        match: /wrapper->rows == Qnil/,
+        replacement: 'rb_tr_managed_from_handle(wrapper->rows) == Qnil'
+      },
+      {
+        match: /wrapper->rows = rb_ary_new\(\);/,
+        replacement: 'wrapper->rows = rb_tr_managed_from_handle(rb_ary_new());'
+      },
+      {
+        match: /rb_yield\(rb_ary_entry\(wrapper->rows, i\)\);/,
+        replacement: 'rb_yield(rb_ary_entry(rb_tr_managed_from_handle(wrapper->rows), i));'
+      },
+      {
+        match: /RARRAY_LEN\(wrapper->rows\)/,
+        replacement: 'RARRAY_LEN(rb_tr_managed_from_handle(wrapper->rows))'
+      },
+      {
+        match: /rb_ary_entry\(wrapper->rows,/,
+        replacement: 'rb_ary_entry(rb_tr_managed_from_handle(wrapper->rows),'
+      },
+      {
+        match: /rb_ary_store\(wrapper->rows,/,
+        replacement: 'rb_ary_store(rb_tr_managed_from_handle(wrapper->rows),'
+      },
+      {
+        match: /return wrapper->rows;/,
+        replacement: 'return rb_tr_managed_from_handle(wrapper->rows);'
+      },
+      {
+        match: /wrapper->rows = rb_ary_new2\(wrapper->numberOfRows\);/,
+        replacement: 'wrapper->rows = rb_tr_handle_for_managed(rb_ary_new2(wrapper->numberOfRows));'
+      },
+      {
+        match: /args\.db_timezone = db_timezone;/,
+        replacement: 'args.db_timezone = rb_tr_handle_for_managed_leaking(db_timezone);'
+      },
+      {
+        match: /args\.app_timezone = app_timezone;/,
+        replacement: 'args.app_timezone = rb_tr_handle_for_managed_leaking(app_timezone);'
+      },
+      {
+        match: /args\.block_given = block;/,
+        replacement: 'args.block_given = rb_tr_handle_for_managed_leaking(block);'
+      },
+      {
+        match: /wrapper->fields = Qnil/,
+        replacement: 'wrapper->fields = rb_tr_handle_for_managed(Qnil)'
+      },
+      {
+        match: /wrapper->rows = Qnil/,
+        replacement: 'wrapper->rows = rb_tr_handle_for_managed(Qnil)'
+      },
+      {
+        match: /wrapper->encoding = encoding/,
+        replacement: 'wrapper->encoding = rb_tr_handle_for_managed(encoding)'
+      },
+      {
+        match: /wrapper->client = client/,
+        replacement: 'wrapper->client = rb_tr_handle_for_managed(client)'
+      },
+      {
+        match: /wrapper->statement = statement/,
+        replacement: 'wrapper->statement = rb_tr_handle_for_managed_leaking(statement)'
+      },
+      {
+        match: /args->app_timezone/,
+        replacement: 'rb_tr_managed_from_handle(args->app_timezone)'
+      },
+      {
+        match: /args->db_timezone/,
+        replacement: 'rb_tr_managed_from_handle(args->db_timezone)'
+      }
```

Comment:
Could you try to make generic regexps for reading and writing a field, parametrizable by the name of the LHS and the field name?
Something like
```ruby
def read(struct_var_name, field_name)
  /\b#{struct_var_name}->#{field_name}\s*([\),;]|==)/
end
```
but it needs to be improved :)

Reply:
(and same for writing to a field)

Reply:
Updated to generic regexps for reading and writing the fields.